_P201_VBAN.ino

```diff
@@ -0,0 +1,577 @@
+  strcpy_P(header->streamName , Settings.Name );/* stream name max 16 chars*/         
+  header->nuFrame = 0;       /* growing frame number. */     
+}
+
+void P201_setFrameCounter(uint8_t *header, uint32_t counter){
+ *(header+24) = (uint8_t)counter; //LSB
+ *(header+25) = (uint8_t)(counter>>8);
+ *(header+26) = (uint8_t)(counter>>16);
+ *(header+27) = (uint8_t)(counter>>24); //MSB
+}
+
+#define ismulticast(addr1) (((uint32_t)(addr1) & PP_HTONL(0xf0000000UL)) == PP_HTONL(0xe0000000UL))
+
+
+int P201_sendUDP(IPAddress toIP, uint16_t port, uint8_t * package, size_t length){
+  if (WiFiConnected()) 
+  {
+    if (port < 0 || port > 65535) return false;
+    if (ismulticast(toIP))
+    {
+      if (p201UDP.beginMulticastPacket()) //(toIP, port, WiFi.localIP(),2))
+      {
+        p201UDP.write(package,length);
+        p201UDP.endPacket();
+      } 
+
+    }
+    else
+      if (p201UDP.beginPacket(toIP, port))
+      {
+        p201UDP.write(package,length);
+        p201UDP.endPacket();
+      }
+    return 0;
+    
+  }
+  return -1;
+}
+
+
+
+//A plugin has to implement the following function
+boolean Plugin_201(byte function, struct EventStruct *event, String& string)
+{
+  //function: reason the plugin was called
+  //event: ??add description here??
+  //string: ??add description here??
+
+  boolean success = false;
+
+  switch (function)
+  {
+    case PLUGIN_DEVICE_ADD:
+    {
+        //This case defines the device characteristics, edit appropriately
+        Device[++deviceCount].Number = PLUGIN_ID_201;
+        Device[deviceCount].Type = DEVICE_TYPE_ANALOG;  //ANALOG-IN , AR (optional), GAIN (optiona)
+        Device[deviceCount].VType = SENSOR_TYPE_SINGLE; //type of value the plugin will return, used only for Domoticz
+        Device[deviceCount].Ports = 0;
+        Device[deviceCount].PullUpOption = false;
+        Device[deviceCount].InverseLogicOption = false;
+        Device[deviceCount].FormulaOption = false;
+        Device[deviceCount].ValueCount = 1;             //number of output variables. The value should match the number of keys PLUGIN_VALUENAME1_xxx
+        Device[deviceCount].SendDataOption = true;
+        Device[deviceCount].TimerOption = true;
+        Device[deviceCount].TimerOptional = true;
+        Device[deviceCount].GlobalSyncOption = true;
+        Device[deviceCount].DecimalsOnly = true;
+        break;
+    }
+
+    case PLUGIN_GET_DEVICENAME:
+    {
+      //return the device name
+      string = F(PLUGIN_NAME_201);
+      break;
+    }
+
+    case PLUGIN_GET_DEVICEVALUENAMES:
+    {
+      //called when the user opens the module configuration page
+      //it allows to add a new row for each output variable of the plugin
+      strcpy_P(ExtraTaskSettings.TaskDeviceValueNames[0], PSTR(PLUGIN_VALUENAME1_201));
+      break;
+    }
+
+    case PLUGIN_SET_DEFAULTS:
+    {
+      PCONFIG(0)=VBAN_PORT;
+      
+      PCONFIG_LONG(0) = IPAddress(192,168,100,62);

```

Comment:
Why this specific default?
It is probably a proper default on your network, but very likely not for anyone else.

Reply:
you are right. will remove that default.

Reply:
the plan is to have a multicast address generated based on current IP later on.

Reply:
Like I said in the other reply, I would be very surprised if multicast would ever work well via WiFi.
The way how WiFi operates just isn't suitable for multicast.
What could eventually work is ESP => AP => wired => wired clients receiving multicast traffic.
But that's probably also very implementation specific for the AP you're using.  (its firmware)

Reply:
Yes, Multicast depends on the AP firmware implementation. 
I do use dd-wrt for my routers.